packages/components/organisms/f-checkout/src/components/Checkout.vue

```diff
@@ -547,8 +541,10 @@ export default {
                     logMethod: this.$logger.logInfo
                 });
             } catch (e) {
-                this.toggleDialogError();
-                throw new PlaceOrderError(e.message);
+                const { errorCode } = e.response.data;
+                const { shouldShowInDialog } = checkoutIssues[errorCode];
+
+                throw new PlaceOrderError(e.message, shouldShowInDialog);
```

Comment:
To future-proof this slightly, and prevent having to add more parameters into the constructor signature as more requirements emerge, I'd recommend passing in the `errorCode` rather than the `shouldShowInDialog` value, and resolve to properties within the constructor by importing `checkout-issues.js` within the exceptions definition file.

Reply:
I'm not sure I agree with this, because now the custom errors have a direct dependency on `checkout-issues.js`.

Reply:
I see what you mean @SachaZvetelman - but we're adding the dependency in here instead.

In that case, given we're already depending on `checkout-issues.js` within the checkout vuex module, would it not be better generally to have try-catches within the vuex actions that result in these errors, and have the store module instantiate and throw the exceptions directly? That keeps the dependency in the store only and slims up the component as no try-catch-rethrow events need to be performed there.